Remoting timeout using remotable logger

The following error sporadically happens when building on a heavily loaded build machine :

```
2018-07-06T14:14:58.2972521Z ##[error]Agent01\358\.nuget\uno.sourcegenerationtasks\1.22.0\build\MonoAndroid\Uno.SourceGenerationTasks.targets(98,4): Error : Generation failed for Uno.UI.SourceGenerators.XamlGenerator.XamlCodeGenerator. System.AggregateException: An error occurred while writing to logger(s). ---> System.Runtime.Remoting.RemotingException: Object '/3011ddec_8e68_49d3_95a0_e339e9cb548f/4editpvjty4l6fc76dzgfnxk_43.rem' has been disconnected or does not exist at the server.
   at Uno.SourceGeneratorTasks.Logger.RemotableLogger2.WriteLog(Int32 logLevel, String message)
   at Uno.SourceGeneratorTasks.Logger.RemoteLogger.Log[TState](LogLevel logLevel, EventId eventId, TState state, Exception exception, Func`3 formatter) in C:\projects\uno-sourcegeneration\src\Uno.SourceGeneratorTasks.Shared\Logger\RemoteLogger.cs:line 71
   at Microsoft.Extensions.Logging.Logger.Log[TState](LogLevel logLevel, EventId eventId, TState state, Exception exception, Func`3 formatter)
   --- End of inner exception stack trace ---
   at Microsoft.Extensions.Logging.Logger.Log[TState](LogLevel logLevel, EventId eventId, TState state, Exception exception, Func`3 formatter)
   at Microsoft.Extensions.Logging.Logger`1.Microsoft.Extensions.Logging.ILogger.Log[TState](LogLevel logLevel, EventId eventId, TState state, Exception exception, Func`3 formatter)
   at Uno.SourceGeneratorTasks.Helpers.LogExtensions.Debug(ILogger log, String message, Exception exception) in C:\projects\uno-sourcegeneration\src\Uno.SourceGeneratorTasks.Shared\Helpers\LogExtensions.cs:line 94
   at Uno.SourceGeneration.Host.SourceGeneratorHost.<>c__DisplayClass2_0.<Generate>b__13(ValueTuple`2 generatorDef) in C:\projects\uno-sourcegeneration\src\Uno.SourceGenerationHost.Shared\SourceGeneratorHost.cs:line 150
---> (Inner Exception #0) System.Runtime.Remoting.RemotingException: Object '/3011ddec_8e68_49d3_95a0_e339e9cb548f/4editpvjty4l6fc76dzgfnxk_43.rem' has been disconnected or does not exist at the server.
   at Uno.SourceGeneratorTasks.Logger.RemotableLogger2.WriteLog(Int32 logLevel, String message)
   at Uno.SourceGeneratorTasks.Logger.RemoteLogger.Log[TState](LogLevel logLevel, EventId eventId, TState state, Exception exception, Func`3 formatter) in C:\projects\uno-sourcegeneration\src\Uno.SourceGeneratorTasks.Shared\Logger\RemoteLogger.cs:line 71
   at Microsoft.Extensions.Logging.Logger.Log[TState](LogLevel logLevel, EventId eventId, TState state, Exception exception, Func`3 formatter)
```

--- a/src/Uno.SourceGenerationHost.Shared/Helpers/TaskLogger.cs
+++ b/src/Uno.SourceGenerationHost.Shared/Helpers/TaskLogger.cs
@@ -31,7 +31,14 @@ namespace Uno.SourceGeneratorTasks.Helpers
             _categoryName = categoryName;
 		}
 
-        public IDisposable BeginScope<TState>(TState state) => new DisposableAction(() => { });
+		public override object InitializeLifetimeService()
+		{
+			// Keep this object alive infinitely, it will be deleted along with the 
+			// host msbuild.exe process.
+			return null;
+		}
+
+		public IDisposable BeginScope<TState>(TState state) => new DisposableAction(() => { });
 
         public bool IsEnabled(LogLevel logLevel) => logLevel >= LogLevel.Debug;
 

--- a/src/Uno.SourceGeneratorTasks.Shared/Logger/RemotableLogger2.cs
+++ b/src/Uno.SourceGeneratorTasks.Shared/Logger/RemotableLogger2.cs
@@ -30,7 +30,14 @@ namespace Uno.SourceGeneratorTasks.Logger
             _log = log;
         }
 
-        public void WriteLog(int logLevel, string message)
+		public override object InitializeLifetimeService()
+		{
+			// Keep this object alive infinitely, it will be deleted along with the 
+			// host msbuild.exe process.
+			return null;
+		}
+
+		public void WriteLog(int logLevel, string message)
         {
 			try
 			{

--- a/src/Uno.SourceGeneratorTasks.Shared/Logger/RemoteLogger.cs
+++ b/src/Uno.SourceGeneratorTasks.Shared/Logger/RemoteLogger.cs
@@ -32,7 +32,14 @@ namespace Uno.SourceGeneratorTasks.Logger
             _loggerName = loggerName;
         }
 
-        public RemotableLogger2 TaskLog
+		public override object InitializeLifetimeService()
+		{
+			// Keep this object alive infinitely, it will be deleted along with the 
+			// host msbuild.exe process.
+			return null;
+		}
+
+		public RemotableLogger2 TaskLog
         {
             get
             {

